### ISSUE TITLE
Overlap between Magicoder-Evol-Instruct-110K and HumanEval

### ISSUE THREAD

**Issue by Truth-In-Lies** [ise-uiuc/magicoder#40]
I've discovered a significant overlap between the Magicoder-Evol-Instruct-110K and HumanEval. Specifically, I found that:

- 82 out of 164 HumanEval problems are directly present in the training set(code snippets can directly pass the unittest of problem in HumanEval)
- Around 130 problems have significant similarities

This is a conservative estimate, and there may be more similar items that I missed.  Approximately 3000 items in the training set resemble those in the test set.

Is the overlap between Magicoder-Evol-Instruct-110K and HumanEval a test set leakage issue, or is it acceptable since the overlapping problems are paraphrased rather than identical?

| HumenEval/ID  | Lineno in MagiCoder-Evol-Instruct-110K  |
|---|---|
|HumanEval/0|60301|
|HumanEval/1|12556|
|HumanEval/3|8985|
|HumanEval/5|1934|
|HumanEval/6|75548|
|HumanEval/8|50011|
|HumanEval/9|99748|
|HumanEval/11|16654|
|HumanEval/12|21800|
|HumanEval/13|2630|
|HumanEval/15|106409|
|HumanEval/17|1993|
|HumanEval/18|54|
|HumanEval/20|76613|
|HumanEval/21|44962|
|HumanEval/24|60016|
|HumanEval/25|52864|
|HumanEval/26|38476|
|HumanEval/27|59049|
|HumanEval/29|77775|
|HumanEval/31|33|
|HumanEval/33|13447|
|HumanEval/34|47957|
|HumanEval/35|108174|
|HumanEval/40|34303|
|HumanEval/47|2879|
|HumanEval/48|103|
|HumanEval/51|17387|
|HumanEval/52|19399|
|HumanEval/55|5484|
|HumanEval/57|778|
|HumanEval/58|4614|
|HumanEval/59|10281|
|HumanEval/60|45722|
|HumanEval/61|12849|
|HumanEval/63|665|
|HumanEval/64|9014|
|HumanEval/66|16677|
|HumanEval/67|66099|
|HumanEval/71|87392|
|HumanEval/72|17223|
|HumanEval/75|107027|
|HumanEval/86|95562|
|HumanEval/87|47935|
|HumanEval/90|4369|
|HumanEval/96|6696|
|HumanEval/97|39228|
|HumanEval/98|5013|
|HumanEval/100|39904|
|HumanEval/102|75537|
|HumanEval/105|12560|
|HumanEval/106|835|
|HumanEval/110|14161|
|HumanEval/116|21351|
|HumanEval/118|7984|
|HumanEval/121|60649|
|HumanEval/137|29729|
|HumanEval/149|20001|
|HumanEval/152|32034|
|HumanEval/154|34346|
|HumanEval/155|41634|
|HumanEval/156|11264|
|HumanEval/158|19315|
|HumanEval/160|105483|
|HumanEval/162|68840|
|HumanEval/80|29996|
|HumanEval/82|72856|
|HumanEval/4|15300|
|HumanEval/22|99988|
|HumanEval/37|91008|
|HumanEval/46|56824|
|HumanEval/70|68607|
|HumanEval/74|57346|
|HumanEval/78|105847|
|HumanEval/89|78000|
|HumanEval/99|76033|
|HumanEval/104|51604|
|HumanEval/109|5608|
|HumanEval/111|44999|
|HumanEval/119|105874|
|HumanEval/148|11486|
|HumanEval/159|80388|

**Comment by UniverseFly** [comment]
Hi @Truth-In-Lies, thanks for your comprehensive analysis, which is very valuable. Here is how we view the contamination problem: since it is intrinsically challenging to formulate a precise definition of contamination (https://arxiv.org/abs/2311.04850), and as you pointed out -- paraphrasing, our decontamination pipeline exactly follows existing good practices, i.e., StarCoder to just remove exact match.

To truly understand the model performance, our approach is to **evaluate the models on as more benchmarks as possible**. I also want to mention two additional and independent evaluation efforts, [LiveCodeBench](https://livecodebench.github.io/) -- a contamination free benchmark of fresh coding tasks, and [EvoEval](https://evo-eval.github.io/) -- evolving existing benchmarks to formulate new coding problems. They all showed Magicoder's good performance.

To sum up, we believe a comprehensive evaluation is the way to resolve the contamination issue and it is possible as you said the model may overfit to one specific benchmark suite due to paraphrasing (e.g., HumanEval).

Hope this answers your question!

**Comment by Truth-In-Lies** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Comment by Truth-In-Lies** [comment]
Thanks so much for your answer and for your amazing work. I think that HumanEval (and MBPP) has been widely leaked on the web, which makes it less useful as a benchmark. I've noticed that many models tend to generate problems similar to those in HumanEval. Maybe it's time for new benchmarks to take their place. Thanks again for your time and expertise!
